rockcraft.yaml

```diff
@@ -87,9 +87,50 @@ parts:
     stage:
       - opt/spark/jars
 
-  user-setup:
+  client-scripts:
     plugin: nil
     after: [hadoop-jars]
+    build-packages:
+      - wget
+      - pip
+    overlay-script: |
+      SPARK_CLIENT_PKG_VERSION='0.0.2'
+      mkdir -p $CRAFT_PART_INSTALL/opt/spark/bin
+      cd $CRAFT_PART_INSTALL/opt/spark/bin
+      echo "Downloading latest available Spark wheel package release v${SPARK_CLIENT_PKG_VERSION}."
+      wget https://github.com/canonical/spark-client-snap/releases/download/v${SPARK_CLIENT_PKG_VERSION}/spark_client-${SPARK_CLIENT_PKG_VERSION}-py3-none-any.whl --no-check-certificate
+      if  [[ $? -ne 0 ]]
+        then
+          echo "DOWNLOAD ERROR: Spark client package v${SPARK_CLIENT_PKG_VERSION} could not be downloaded properly! Exiting...."
+          exit 1
+      fi
+      pip install ./spark_client-${SPARK_CLIENT_PKG_VERSION}-py3-none-any.whl
```

Comment:
This seems to be installing the package in the rock build environment and not packaging the downloaded scripts to the rock itself. Needs to be figured out how to achieve this elegantly.

Reply:
Also this has to be available for use i.e. accessible for a non root user.